proxy/contracts/mainnet/DepositBoxes/DepositBoxERC20.sol

```diff
@@ -288,12 +510,137 @@ contract DepositBoxERC20 is DepositBox, IDepositBoxERC20 {
+            DelayedTransferStatus status = transfer.status;
+            if (transfer.token == token) {
+                if (status == DelayedTransferStatus.DELAYED) {
+                    unlockTimestamp = MathUpgradeable.min(unlockTimestamp, transfer.untilTimestamp);
+                    break;
+                } else if (status == DelayedTransferStatus.ARBITRAGE) {
+                    unlockTimestamp = MathUpgradeable.min(unlockTimestamp, transfer.untilTimestamp);
+                }
+            }
+        }
+    }
+
+    /**
+     * @dev Get amount of addresses that are added to the whitelist
+     */
+    function getTrustedReceiversAmount(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].trustedReceivers.length();
+    }
+
+    /**
+     * @dev Get i-th address of the whitelist
+     */
+    function getTrustedReceiver(string calldata schainName, uint256 index) external view override returns (address) {
+        return _delayConfig[_schainHash(schainName)].trustedReceivers.at(index);
+    }
+
+    /**
+     * @dev Get amount of tokens that are considered as a big transfer
+     */
+    function getBigTransferThreshold(bytes32 schainHash, address token) external view override returns (uint256) {
+        return _delayConfig[schainHash].bigTransferThreshold[token];
+    }
+
+    /**
+     * @dev Get time delay of big transfers
+     */
+    function getTimeDelay(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].transferDelay;
+    }
+
+    /**
+     * @dev Get duration of an arbitrage
+     */
+    function getArbitrageDuration(bytes32 schainHash) external view override returns (uint256) {
+        return _delayConfig[schainHash].arbitrageDuration;
+    }
+
+    /**
+     * @dev Retrive tokens that were unlocked after delay for specified receiver
+     */
+    function retrieveFor(address receiver) public override {
+        uint256 transfersAmount = MathUpgradeable.min(
+            delayedTransfersByReceiver[receiver].length(),
+            _QUEUE_PROCESSING_LIMIT
+        );
+
+        uint256 currentIndex = 0;
+        for (uint256 i = 0; i < transfersAmount; ++i) {
+            uint256 transferId = uint256(delayedTransfersByReceiver[receiver].at(currentIndex));
+            DelayedTransfer memory transfer = delayedTransfers[transferId];            
+            ++currentIndex;
+
+            if (transfer.status != DelayedTransferStatus.COMPLETED) {
+                if (block.timestamp < transfer.untilTimestamp) {
+                    // disable detector untill slither fixes false positive
+                    // https://github.com/crytic/slither/issues/778
+                    // slither-disable-next-line incorrect-equality
+                    if (transfer.status == DelayedTransferStatus.DELAYED) {
+                        break;
```

Comment:
I would like to have a revert with some error like "Nothing to retrieve" instead of execute tx and get nothing

Reply:
Agreed.